examples/mosquitto/README.md

```diff
@@ -0,0 +1,225 @@
+```
+server {
+    bind_address = "10.211.55.2"
+    bind_port = "8081"
+    socket_path = "/tmp/spire-server/private/api.sock"
+    trust_domain = "example.org"
+    data_dir = "./.data"
+    log_level = "DEBUG"
+}
+
+plugins {
+    DataStore "sql" {
+        plugin_data {
+            database_type = "sqlite3"
+            connection_string = "./.data/datastore.sqlite3"
+        }
+    }
+
+    NodeAttestor "join_token" {
+        plugin_data {
+        }
+    }
+
+    KeyManager "memory" {
+        plugin_data = {}
+    }
+
+    UpstreamAuthority "disk" {
+        plugin_data {
+            key_file_path = "./conf/server/dummy_upstream_ca.key"
+            cert_file_path = "./conf/server/dummy_upstream_ca.crt"
+        }
+    }
+}
+```
+
+Start the server with:
+```bash
+cd /opt/spire
+./bin/spire-server run -config conf/server/server.conf
+```
+
+*NOTE:* Make sure `/opt/spire` directory has the right permissions for the
+server to write to it.
+
+### 2. Configure and Run Spire Agent on Node1 & Node2
+
+Just like with the server, install Spire under `/opt/spire` and edit the agent
+configuration file to set the `bind_address` to the IP of the spire server (i.e.
+node0) and the `trust_domain` to `example.org`. The rest of the configuration
+can be left as is.
+
+Before we can start the agents up on `node1` and `node2` we need to register
+them with the server.
+
+### 3. Register Node1 Agent
+
+On `node0` run the following command:
+```bash
+./bin/spire-server token generate -spiffeID spiffe://example.org/node1
+```
+This will output a join token. Copy it and run the following command on `node1`:
+```bash
+./bin/spire-agent run -config ./conf/agent/agent.conf -joinToken ${JOIN_TOKEN}
+```
+Make sure to replace `${JOIN_TOKEN}` with the token you got from the server.
+
+### 4. Register Node2 Agent
+
+On `node0` run the following command:
+```bash
+./bin/spire-server token generate -spiffeID spiffe://example.org/node2
+```
+This will output a join token. Copy it and run the following command on `node2`:
+```bash
+./bin/spire-agent run -config ./conf/agent/agent.conf -joinToken ${JOIN_TOKEN}
+```
+Make sure to replace `${JOIN_TOKEN}` with the token you got from the server.
+
+### 5. Install Mosquitto Broker on Node1
+
+Install latest [Mosquitto](https://mosquitto.org/) broker on `node1`. You can
+best use the package that comes with the distribution. On Ubuntu this is done by
+running:
+```bash
+sudo apt install mosquitto
+```
+
+Once installed, make sure to stop it as we will be running it with the spiffe
+helper instead:
+```bash
+sudo systemctl stop mosquitto
+```
+
+### 6. Register Mosquitto Broker Workload with Spire Server
+
+On `node0` run the following command:
+```bash
+cd /opt/spire
+./bin/spire-server entry create \
+                   -spiffeID spiffe://example.org/mosquitto-server \
+                   -parentID spiffe://example.org/node1 \
+                   -selector unix:user:mosquitto \
+                   -ttl 600 \
```

Comment:
If the ttl is not strictly necessary i would leave it out and let it be default value.

Reply:
It is similar to the PostgreSQL sample that also adjusts the ttl to show the certificate expiry. I think it is a nice touch to clearly show the benefit of using spiffe. Maybe I should make it clearer that the ttl is purely for testing and in production you would want to keep it the default or some larger sensible value?

Reply:
Ok, i think its fine to leave.